### PR TITLE
Celery implemented.

### PR DESCRIPTION
**Pull request by Limpan**
There's a bunch of environment variables added. See https://github.com/teknik-eksjo/chronos/wiki/.env-file

### CODE REVIEW

**Comment by Greenheart** [comment]
You must mean "Celery **integrated**"? :wink:

**Comment by Limpan** [comment]
Semantics...

**Comment by Greenheart** [comment]
I think this is ready to be merged. Let's merge it tomorrow and spend 10-15 minutes to walk it through and modify our `.env`-files.

**Comment by hugolundin** [comment]
Maybe it is a good idea to test it on another computer before we merge it - if I recall correctly, we had some problems on mac last time we tried Celery 🌵

**Comment by Greenheart** [comment]
@hugolundin Good idea! Can you verify this tomorrow? :smile:

**Comment by hugolundin** [comment]
Yep 👍

**Comment by Greenheart** [comment]
This seems to need a rebase before we can merge it.

@hugolundin Have you reviewed on mac? :blush:

**Comment by Limpan** [comment]
Rebased. There we're merge conflicts so @hugolundin and others might want to test it once again. Just to make sure.

**Comment by hugolundin** [comment]
Will do asap

**Comment by Greenheart** [comment]
I've successfully rebuilt my container to work with these changes. Is there anything special I should test?

File: web/app/tasks.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+logger = get_task_logger(__name__)
+
+
+@celery.task
+def do_something(user_id):
+    try:
+        user = User.query.filter_by(id=user_id).one()
+
+        # Do something useful...
+        logger.info('Fetched user {}'.format(user.id or '-'))
```
Comment (Greenheart) [inline web/app/tasks.py:18]: This seems to be unused at the moment, or where can I find the logger-output?

**Comment by Limpan** [comment]
`./manage.py shell`
`from app.tasks import do_something`
`t = do_something.delay(1)`
`print(t.status)`

Should work and will tell you the status of the task. The logging might be possible to see with `docker-compose logs -f chronos_worker_1`. It might also be that logging isn't set up properly and it disappears.

`delay()` passes almost all of it's arguments through to `do_something`.

**Comment by hugolundin** [comment]
It worked for me. Ready to merge? @teknik-eksjo/te13

**Comment by Greenheart** [comment]
@Limpan The rest of the application works fine but the commands you wrote n your latest comment doesn't work. Maybe there's some problem with my container(s)?

```
(venv) sam@ubbe:~/Documents/github/chronos/web$ ./manage.py shell
Importing environment from .env...
>>> from app.tasks import do_something
>>> t = do_something.delay(1)
Traceback (most recent call last):
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/utils/__init__.py", line 423, in __call__
    return self.__value__
AttributeError: 'ChannelPromise' object has no attribute '__value__'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/connection.py", line 436, in _ensured
    return fun(*args, **kwargs)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/messaging.py", line 177, in _publish
    channel = self.channel
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/messaging.py", line 194, in _get_channel
    channel = self._channel = channel()
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/utils/__init__.py", line 425, in __call__
    value = self.__value__ = self.__contract__()
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/messaging.py", line 209, in <lambda>
    channel = ChannelPromise(lambda: connection.default_channel)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/connection.py", line 757, in default_channel
    self.connection
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/connection.py", line 742, in connection
    self._connection = self._establish_connection()
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/connection.py", line 697, in _establish_connection
    conn = self.transport.establish_connection()
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/transport/pyamqp.py", line 116, in establish_connection
    conn = self.Connection(**opts)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/amqp/connection.py", line 165, in __init__
    self.transport = self.Transport(host, connect_timeout, ssl)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/amqp/connection.py", line 186, in Transport
    return create_transport(host, connect_timeout, ssl)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/amqp/transport.py", line 299, in create_transport
    return TCPTransport(host, connect_timeout)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/amqp/transport.py", line 75, in __init__
    socket.SOCK_STREAM, SOL_TCP):
  File "/usr/lib/python3.5/socket.py", line 732, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/celery/app/task.py", line 453, in delay
    return self.apply_async(args, kwargs)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/celery/app/task.py", line 565, in apply_async
    **dict(self._get_exec_options(), **options)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/celery/app/base.py", line 354, in send_task
    reply_to=reply_to or self.oid, **options
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/celery/app/amqp.py", line 310, in publish_task
    **kwargs
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/messaging.py", line 172, in publish
    routing_key, mandatory, immediate, exchange, declare)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/connection.py", line 457, in _ensured
    interval_max)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/connection.py", line 369, in ensure_connection
    interval_start, interval_step, interval_max, callback)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/utils/__init__.py", line 246, in retry_over_time
    return fun(*args, **kwargs)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/connection.py", line 237, in connect
    return self.connection
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/connection.py", line 742, in connection
    self._connection = self._establish_connection()
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/connection.py", line 697, in _establish_connection
    conn = self.transport.establish_connection()
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/kombu/transport/pyamqp.py", line 116, in establish_connection
    conn = self.Connection(**opts)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/amqp/connection.py", line 165, in __init__
    self.transport = self.Transport(host, connect_timeout, ssl)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/amqp/connection.py", line 186, in Transport
    return create_transport(host, connect_timeout, ssl)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/amqp/transport.py", line 299, in create_transport
    return TCPTransport(host, connect_timeout)
  File "/home/sam/Documents/github/chronos/web/venv/lib/python3.5/site-packages/amqp/transport.py", line 75, in __init__
    socket.SOCK_STREAM, SOL_TCP):
  File "/usr/lib/python3.5/socket.py", line 732, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known
>>> print(t.status)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 't' is not defined
>>> 

```

**Comment by Limpan** [comment]
What values do you environment variables `BROKER_URL` and `RESULT_BACKEND` have?

You can also check for output in your container logs.

**Comment by Greenheart** [comment]
@Limpan Oh, I forgot to change the `.env`-file :smile: 
I think this is ready to merge!

**Comment by Greenheart** [comment]
@teknik-eksjo/te13 Does anyone have additional feedback? Otherwise let's merge this :)

**Comment by hugolundin** [comment]
I think it seems to be ready! Gogo 🔂

**Comment by Greenheart** [comment]
@teknik-eksjo/te13 Remember to update your `.env`-files!

**Pull request merged**